src/compile/mark/encode/position-rect.ts

```diff
@@ -220,22 +220,19 @@ function getBinSpacing(
 
   const spacingOffset = channel === 'x' || channel === 'y2' ? -spacing / 2 : spacing / 2;
 
-  if (isSignalRef(reverse)) {
+  if (isSignalRef(reverse) || isSignalRef(offset) || isSignalRef(translate)) {
```

Comment:
simplify the logic of this method by collapsing into 2 cases:

1. If there is a signal for any of `reverse`, `offset`, or `translate`
2. If none are signals.

Tests are modified to reflect this change.